modules/querier/http.go

```diff
@@ -92,30 +95,37 @@ func (q *Querier) TraceByIDHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// return values are (valid, blockStart, blockEnd, queryIngesters)
-func validateAndSanitizeRequest(r *http.Request) (string, string, bool, error) {
-	// get parameter values
-	q := r.URL.Query().Get(QueryIngestersKey)
-	start := r.URL.Query().Get(BlockStartKey)
-	end := r.URL.Query().Get(BlockEndKey)
-
-	// validate queryIngesters. it should either be empty or one of (true|false)
-	var queryIngesters bool
-	if len(q) == 0 || q == "true" {
-		queryIngesters = true
-	} else if q == "false" {
-		queryIngesters = false
+// return values are (blockStart, blockEnd, queryMode, error)
+func validateAndSanitizeRequest(r *http.Request) (string, string, string, error) {
```

Comment:
mode is a breaking change and should be noted.  i believe the only impact will be some rockiness on rollout.